packages/runtime/datastore-definitions/src/jsonable.ts

```diff
@@ -3,19 +3,54 @@
  * Licensed under the MIT License.
  */
 
+/**
+ * Type constraint for types that are likely serializable as JSON or have a custom
+ * alternate type.
+ *
+ * @remarks
+ * Use `JsonableTypeWith<never>` for just JSON serializable types.
+ * See {@link Jsonable} for serialization pitfalls.
+ */
+export type JsonableTypeWith<T> =
+	| undefined
+	| null
+	| boolean
+	| number
+	| string
+	| T
+	| Internal_InterfaceOfJsonableTypesWith<T>
+	| ArrayLike<JsonableTypeWith<T>>;
```

Comment:
We should clarify when to use this type vs. other exported types.

(The simplest story is to say this is an implementation detail and point people at Jsonable.)